pkg/blobstore/configuration/new_blob_access.go

```diff
@@ -239,9 +240,15 @@ func newNestedBlobAccessBare(configuration *pb.BlobAccessConfiguration, creator
 					DialTimeout:     dialTimeout,
 					ReadTimeout:     readTimeout,
 					WriteTimeout:    writeTimeout,
+					NewClient: func(opt *redis.Options) *redis.Client {
```

Comment:
If you were to place this function into a free function somewhere at the top of the file, you could also call into it below to construct the single Redis client, right?

Wouldn't it make more sense to make this change as part of the OpenTelemetry PR?

Reply:
The OpenTelemetry PR doesn't have a redis version which includes redisext.OpenTelemetryHook, so there isn't an obviously correct ordering here.

I can split the function out, although it then ends up quite distant from this code. (Tempting to split up new_blob_access, it's something of a skyscraper)